mypy/checker.py

```diff
@@ -186,6 +187,13 @@ def visit_file(self, file_node: MypyFile, path: str) -> None:
                 self.fail(messages.ALL_MUST_BE_SEQ_STR.format(str_seq_s, all_s),
                           all_.node)
 
+    @contextmanager
+    def new_scope(self) -> Iterator[ConditionalTypeBinder]:
+        old_binder = self.binder
+        self.binder = ConditionalTypeBinder()
```

Comment:
Oh yeah. A certain quiz is coming back to me now... Let's just leave this alone for the moment.